ie/source/Logger.cpp

```diff
@@ -112,30 +97,275 @@ void Logger::write(const std::wstring& message, Logger::Level level)
+        this->logAllEnums(hKey);
+    } catch (...) {
+        this->error(L"ERROR: Couldn't log security sites");
+    }
+}
+
+void Logger::logAllEnums(HKEY hKey)
+{
+    DWORD index = 0;
+    TCHAR keyName[256] = { 0 };
+    DWORD keyLen = 256;
+    while (::RegEnumKeyEx(hKey, index++, keyName, &keyLen, 0, 0, 0, 0) == ERROR_SUCCESS)
+    {
+        std::wstring strLog = L"Domain: ";
+        strLog.append(keyName);
+        this->logSystem(strLog);
+        keyLen = 256;
+        HKEY hSubKey = { 0 };
+        if (::RegOpenKeyEx(hKey, keyName, 0, KEY_READ, &hSubKey) == ERROR_SUCCESS)
+        {
+            DWORD dwReturnHttps;
+            DWORD dwReturnHttp;
+            if (this->readRegistryW(hSubKey, TEXT("https"), reinterpret_cast<LPBYTE>(&dwReturnHttps)) == ERROR_SUCCESS ||
+                this->readRegistryW(hSubKey, TEXT("http"), reinterpret_cast<LPBYTE>(&dwReturnHttp)) == ERROR_SUCCESS)
+            {
+                bool tbd = true;
+                if (dwReturnHttps == INTRANET) {
+                    this->logSystem(L"-> intranet HTTPs");
+                    tbd = false;
+                }
+                else if (dwReturnHttps == TRUSTED)
+                {
+                    this->logSystem(L"-> trusted HTTPs");
+                    tbd = false;
+                }
+                else if (dwReturnHttps == RESTRICTED)
+                {
+                    this->logSystem(L"-> restricted HTTPs");
+                    tbd = false;
+                }
+                if (dwReturnHttp == INTRANET) {
+                    this->logSystem(L"-> intranet HTTPs");
+                    tbd = false;
+                }
+                else if (dwReturnHttp == TRUSTED)
+                {
+                    this->logSystem(L"-> trusted HTTP");
+                    tbd = false;
+                }
+                else if (dwReturnHttp == RESTRICTED)
+                {
+                    this->logSystem(L"-> restricted HTTP");
+                    tbd = false;
+                }
+                if (tbd) {
+                    this->logSystem(L"-> TBD");
+                }
+            }
+            else
+            {
+                this->logSystem(L"- Subs:");
+                this->logAllEnums(hSubKey);
+            }
+
+            ::RegCloseKey(hSubKey);
+        }
+    }
+}
+
+std::wstring Logger::getZoneName(const int zone) {
+    if (zone == INTRANET_ZONE)
+    {
+        return L"Intranet";
+    }
+    else if (zone == TRUSTED_SITES_ZONE)
+    {
+        return L"Trusted Sites";
+    }
+    else if (zone == INTERNET_ZONE)
+    {
+        return L"Internet";
+    }
+    else if (zone == RESTRICTED_SITES_ZONE)
+    {
+        return L"Restricted Sites";
+    }
+
+    return L"Unknown";
+}
+
+LONG Logger::readRegistryW(HKEY hKey, LPCTSTR hValueName, LPBYTE dwReturn)
+{
+    DWORD dwType = REG_DWORD;
+    DWORD dwSize = sizeof(DWORD);
+    return ::RegQueryValueExW(hKey, hValueName, NULL, &dwType, dwReturn, &dwSize);

```

Comment:
the 'dwType' is output parameter. If the ::RegQueryValueExW returns ERROR_SUCCESS then you have to check the type is really REG_DWORD and either pass back ERROR_SUCCESS if that's the case, or some error code if the type is not REG_DWORD